handler/user_handler.go

```diff
@@ -0,0 +1,205 @@
+	return &UserHandler{UserRepository: UserRepo}
+}
+
+func (handler *UserHandler) PostUser(c *gin.Context) {
+	var requestUser = &models.UserRequest{}
+
+	if err := c.BindJSON(&requestUser); err != nil {
+		zap.S().Error("Error: ", zap.Error(err))
+		c.JSON(http.StatusBadRequest, nil)
+		return
+	}
+	if !requestUser.Validate() {
+		zap.S().Error("Error: Paswords do not match")
+		c.JSON(http.StatusBadRequest, nil)
+		return
+	}
+
+	hashed := utils.HashPassword(requestUser.Password)
+
+	var newUser = &entities.User{
+		Email:    requestUser.Email,
+		Password: hashed,
+		Status: "Pending", //test için
+	}
+
+	err := handler.UserRepository.Insert(newUser)
+	if err != nil {
+		zap.S().Error("Error: ", err)
+		c.JSON(http.StatusInternalServerError, nil)
+		return
+	}
+
+	var userResponse = &models.UserResponse{
+		Id:        newUser.Id,
+		Email:     newUser.Email,
+		Status:    newUser.Status,
+		CreatedAt: newUser.CreatedAt,
+		UpdatedAt: newUser.UpdatedAt,
+		DeletedAt: newUser.DeletedAt,
+	}
+
+	c.JSON(http.StatusCreated, userResponse)
+}
+
+func (handler *UserHandler) PatchUser(c *gin.Context) {
+	var user entities.User
+	id, err := strconv.ParseUint(c.Param("user_id"), 10, 64)
+	if err != nil {
+		zap.S().Error("Error: ", zap.Error(err))
+		c.JSON(http.StatusBadRequest, nil)
+		return
+	}
+	user.Id = id
+
+	record, err := handler.UserRepository.FindByID(&user)
+	if err != nil {
+		zap.S().Error("Error: ", zap.Error(err))
+		c.JSON(http.StatusInternalServerError, nil)
+		return
+	}
+	if record == nil {
+		zap.S().Error("Error: ", zap.Error(err))
+		c.JSON(http.StatusNotFound, nil)
+		return
+	}
+
+	var requestUser = &models.UserPatchRequest{}
+	if err := c.BindJSON(&requestUser); err != nil {
+		zap.S().Error("Error: ", zap.Error(err))
+		c.JSON(http.StatusBadRequest, nil)
+		return
+	}
+	if !requestUser.Validate() {
+		zap.S().Error("Error: Paswords do not match")
+		c.JSON(http.StatusBadRequest, nil)
+		return
+	}
+
+	hash := utils.HashPassword(requestUser.Password)
+	user.Password = hash
+
+	err = handler.UserRepository.Update(&user)
+	if err != nil {
+		zap.S().Error("Error: ", zap.Error(err))
+		c.JSON(http.StatusInternalServerError, nil)
+		return
+	}
+
+	var userResponse = &models.UserResponse{
+		Id:        user.Id,
+		Email:     user.Email,
+		Status:    user.Status,
+		CreatedAt: user.CreatedAt,
+		UpdatedAt: user.UpdatedAt,
+		DeletedAt: user.DeletedAt,
+	}
+
+	c.JSON(http.StatusOK, userResponse)
+}
+
+func (handler *UserHandler) DeleteUser(c *gin.Context) {
+	userId, err := strconv.ParseUint(c.Param("user_id"), 10, 64)
+	if err != nil {
+		zap.S().Error("Error: ", zap.Error(err))
+		c.JSON(http.StatusBadRequest, nil)
+		return
+	}
+
+	var deleteUser = &entities.User{
+		Id:     userId,
+		Status: "Passive",
+	}
+
+	err = handler.UserRepository.Update(deleteUser)
+	if err != nil {
+		zap.S().Error("Error: ", zap.Error(err))
+		c.JSON(http.StatusInternalServerError, nil)
+		return
+	}
+
+	err = handler.UserRepository.Delete(deleteUser)
```

Comment:
yüksek ihtimalle soft delete için yaptın ama burda neden hem delete hem update var? senin için kullanıcının status'u passive ise zaten silinmiş oluyor updatedAt de doğal olarak silindiği zaman'a eşitleniyor. yani deletedAt'i kullanmanın bir anlamı yok burada

Reply:
soft delete için yaptım get ettiğinde yine çıkıyordu passive olsa da status hem soft delete olsun hemde bu sorunu çözeyim dedim

Reply:
bunu eklemek yerine condition ekleyebilirsin where("status = ACTIVE") gibi 2 sql sorgusu çalıştırmak yerine. çünkü status tutmak daha mantıklı pending olabilir passive olabilir active olabilir